src/histogram.rs

```diff
@@ -0,0 +1,332 @@
+// Copyright 2014 The Prometheus Authors
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::convert::From;
+use std::sync::{Arc, RwLock};
+use std::collections::HashMap;
+
+use protobuf::RepeatedField;
+use proto;
+use desc::Desc;
+use errors::{Result, Error};
+use value::make_label_pairs;
+use vec::{MetricVec, MetricVecBuilder};
+use metrics::{Collector, Metric, build_fq_name};
+
+const DEFAULT_BUCKETS: &'static [f64; 11] = &[0.005, 0.01, 0.025, 0.05, 0.1, 0.25, 0.5, 1.0, 2.5,
+                                              5.0, 10.0];
+
+// `BUCKET_LABEL` is used for the label that defines the upper bound of a
+// bucket of a histogram ("le" -> "less or equal").
+const BUCKET_LABEL: &'static str = "le";
+
+
+/// `HistogramOpts` bundles the options for creating a Histogram metric. It is
+/// mandatory to set Name and Help to a non-empty string. All other fields are
+/// optional and can safely be left at their zero value.
+pub struct HistogramOpts {
```

Comment:
no need the same with golang client, we can use Opts in HistogramOpts here to simply code.